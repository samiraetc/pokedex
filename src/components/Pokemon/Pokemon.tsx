import React from 'react';
import { PokemonProps } from '../../types/typing';

export const COMPONENT_ID = 'pokedex-pokemon'

const Pokemon = ({ pokemon }: PokemonProps) => {
    return (
            <div className="pokemon" id={COMPONENT_ID}>
                <img src={pokemon.sprites.front_default} alt={pokemon.name} className="pokemon-avatar" />
                <div className='pokemon__infos'>
                    <p className="pokemon__name">{pokemon.name}</p>
                    <p className="pokemon__number">#{pokemon.id}</p>
                </div>
                <div className="pokemon__infos">
                    <p>{pokemon.weight / 10} kg</p>
                    <p>{pokemon.height / 10} m</p>
                </div>
            </div>
    )
}

export default Pokemon;