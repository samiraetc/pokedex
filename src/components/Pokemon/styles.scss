.pokemon {
    font-weight: 700;
    line-height: 19px;

    &-avatar {
        width: 150px;
    }

    &__number {
        font-weight: $pokedex-font-weight-light;
        color: $pokedex-color-blue-strong;
        font-size: $pokedex-font-size-14;
    }

    &__name {
        font-weight: $pokedex-font-weight-bold;
        color: black;
        font-size: $pokedex-font-size-18;
    }

    &__infos {
        margin: $pokedex-spacing-8;
        display: flex;
        align-items: center;
        justify-content: center;
        font-size: $pokedex-font-size-14;
        color: $pokedex-color-yellow-strong;
        gap: $pokedex-spacing-40;
    }
}