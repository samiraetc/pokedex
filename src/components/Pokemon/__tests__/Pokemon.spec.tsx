import { render, configure, fireEvent } from '@testing-library/react'
import Pokemon, { COMPONENT_ID } from '../Pokemon';

configure({
    testIdAttribute: 'id',
});


describe('[POKEMON]', () => {
    test('should render Pokemon', () => {
        const { asFragment, getByText, getByTestId } = render(<Pokemon pokemon={{
            name: 'raticate',
            id: 20,
            abilities: [{ ability: { name: 'run-away' } }
            ],
            sprites: { front_default: 'string' },
            types: [{ type: { name: 'normal' } }],
            stats: [{ base_stat: 20, stat: { name: 'hp' } }],
            weight: 185,
            height: 7,
        }} />);

        expect(getByText('raticate'));
        expect(getByText('#20'));
        expect(getByTestId(COMPONENT_ID)).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

});