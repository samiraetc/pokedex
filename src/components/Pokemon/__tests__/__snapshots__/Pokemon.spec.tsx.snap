// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[POKEMON] should render Pokemon 1`] = `
<DocumentFragment>
  <div
    class="pokemon"
    id="pokedex-pokemon"
  >
    <img
      alt="raticate"
      class="pokemon-avatar"
      src="string"
    />
    <div
      class="pokemon__infos"
    >
      <p
        class="pokemon__name"
      >
        raticate
      </p>
      <p
        class="pokemon__number"
      >
        #20
      </p>
    </div>
    <div
      class="pokemon__infos"
    >
      <p>
        18.5 kg
      </p>
      <p>
        0.7 m
      </p>
    </div>
  </div>
</DocumentFragment>
`;
